plugins/MachineSettingsAction/MachineSettingsExtruderTab.qml

```diff
@@ -0,0 +1,180 @@
+            anchors.left: parent.left
+            width: parent.width * 2 / 3
+
+            spacing: base.columnSpacing
+
+            Label   // Title Label
+            {
+                text: catalog.i18nc("@title:label", "Nozzle Settings")
+                font: UM.Theme.getFont("medium_bold")
+                renderType: Text.NativeRendering
+            }
+
+            Cura.NumericTextFieldWithUnit  // "Nozzle size"
+            {
+                id: extruderNozzleSizeField
+                visible: !Cura.MachineManager.hasVariants
+                containerStackId: base.extruderStackId
+                settingKey: "machine_nozzle_size"
+                settingStoreIndex: propertyStoreIndex
+                labelText: catalog.i18nc("@label", "Nozzle size")
+                labelFont: base.labelFont
+                labelWidth: base.labelWidth
+                controlWidth: base.controlWidth
+                unitText: catalog.i18nc("@label", "mm")
+                forceUpdateOnChangeFunction: forceUpdateFunction
+            }
+
+            Cura.NumericTextFieldWithUnit  // "Compatible material diameter"
+            {
+                id: extruderCompatibleMaterialDiameterField
+                containerStackId: base.extruderStackId
+                settingKey: "material_diameter"
+                settingStoreIndex: propertyStoreIndex
+                labelText: catalog.i18nc("@label", "Compatible material diameter")
+                labelFont: base.labelFont
+                labelWidth: base.labelWidth
+                controlWidth: base.controlWidth
+                unitText: catalog.i18nc("@label", "mm")
+                forceUpdateOnChangeFunction: forceUpdateFunction
+                // Other modules won't automatically respond after the user changes the value, so we need to force it.
+                afterOnEditingFinishedFunction: updateMaterialDiameter
+            }
+
+            Cura.NumericTextFieldWithUnit  // "Nozzle offset X"
+            {
+                id: extruderNozzleOffsetXField
+                containerStackId: base.extruderStackId
+                settingKey: "machine_nozzle_offset_x"
+                settingStoreIndex: propertyStoreIndex
+                labelText: catalog.i18nc("@label", "Nozzle offset X")
+                labelFont: base.labelFont
+                labelWidth: base.labelWidth
+                controlWidth: base.controlWidth
+                unitText: catalog.i18nc("@label", "mm")
+                forceUpdateOnChangeFunction: forceUpdateFunction
+            }
+
+            Cura.NumericTextFieldWithUnit  // "Nozzle offset Y"
+            {
+                id: extruderNozzleOffsetYField
+                containerStackId: base.extruderStackId
+                settingKey: "machine_nozzle_offset_y"
+                settingStoreIndex: propertyStoreIndex
+                labelText: catalog.i18nc("@label", "Nozzle offset Y")
+                labelFont: base.labelFont
```

Comment:
Isn't labelFont something that should be set by the numericTextFieldWith unit? I mean, by default we just use UM.Theme.getFont("default") right?

Reply:
There's a default font in `NumericTextFieldWithUnit`, which is `UM.Theme.getFont("medium")`. I think `default` is probably too small so I configured it here. Because I was tweaking and I wanted to keep the font consistent in the machine settings dialog, I added this `base.labelFont`